psl-core/src/main/java/org/linqs/psl/util/VizDataCollection.java

```diff
@@ -0,0 +1,158 @@
+
+        public VisualizationData() {
+            truthMap = new HashMap<String, Float>();
+            rules = new HashMap<String, Map<String, Object>>();
+            groundRules = new HashMap<String, Map<String, Object>>();
+            groundAtoms = new HashMap<String, Map<String, Object>>();
+        }
+    }
+
+    // Takes in a prediction truth pair and adds it to our map
+    public static void addTruth(GroundAtom target, float truthVal ) {
+        String groundAtomID = Integer.toString(System.identityHashCode(target));
+        vizData.truthMap.put(groundAtomID, truthVal);
+    }
+
+    public static void dissatisfactionPerGroundRule(GroundRuleStore groundRuleStore) {
+        for (GroundRule groundRule : groundRuleStore.getGroundRules()) {
+            String strGroundRuleId = Integer.toString(System.identityHashCode(groundRule));
+            if (groundRule instanceof WeightedGroundRule) {
+                Map<String, Object> groundRuleObj = vizData.groundRules.get(strGroundRuleId);
+                WeightedGroundRule weightedGroundRule = (WeightedGroundRule) groundRule;
+                groundRuleObj.put("dissatisfaction", weightedGroundRule.getIncompatibility());
+            }
+        }
+    }
+
+    // TODO: Arithmetic Ground Rules Collection
+    public static synchronized void ruleMapInsertElement(AbstractLogicalRule parentRule,
+            GroundRule groundRule, Map<Variable, Integer> variableMap,  Constant[] constantsList) {
+        if (groundRule == null) {
+            return;
+        }
+
+        // Adds a groundAtom element to RuleMap
+        ArrayList<Integer> atomHashList = new ArrayList<Integer>();
+        HashSet<GroundAtom> atomSet = new HashSet<GroundAtom>(groundRule.getAtoms());
+        int atomCount = 0;
+        for (GroundAtom a : atomSet) {
+            atomHashList.add(System.identityHashCode(a));
+            Map<String, Object> groundAtomElement = new HashMap<String, Object>();
+            groundAtomElement.put("text", a.toString());
+            groundAtomElement.put("prediction", a.getValue());
+            vizData.groundAtoms.put(Integer.toString(System.identityHashCode(a)), groundAtomElement);
+            atomCount++;
+        }
+
+        // Adds a rule element to RuleMap
+        JSONObject rulesElement = new JSONObject();
+        String ruleStringID = Integer.toString(System.identityHashCode(parentRule));
+        Map<String, Object> rulesElementItem = new HashMap<String, Object>();
+        rulesElementItem.put("text", parentRule.getName());
+        rulesElementItem.put("weighted", parentRule.isWeighted());
+        vizData.rules.put(ruleStringID, rulesElementItem);
+
+        // Adds a groundRule element to RuleMap
+        Map<String, String> varConstMap = new HashMap<String, String>();
+        for (Map.Entry<Variable, Integer> entry : variableMap.entrySet()) {
+            varConstMap.put(entry.getKey().toString(), constantsList[entry.getValue()].rawToString());
+        }
```

Comment:
Insert a newline after the for loop.
You want to break up logical chunks of code with vertical whitespace.